### PR TITLE
Loaction route in admin panel need improvements

### PR DESCRIPTION
**Pull request by kimenyikevin**
## Summary

A summary of changes being made in this PR

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

References to pertaining issue(s)

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

@HexaField @speigg @NateTheGreatt
#4610

## Taks to be completed 
- Search location on admin dashboard 
- Fix bug on update location 
- Define re-usable component in /admin/common (It is only for location, I will continue to update other component)
- Define custom hooks in admin/common/hooks
- Merge style in one file /admin/style/ui

### CODE REVIEW

**Comment by HexaField** [comment]
You need to rebase with the latest xrengine, you are adding a lot of old code back in. Please look at the diff https://github.com/XRFoundation/XREngine/pull/4719/files

**Comment by zulqarnainhanif** [comment]
Delete the social workspace from the local repo that would probably clean up all old code.

File: .gitignore [hunk 1]
```diff
@@ -158,6 +158,7 @@ certs/
 /packages/projects/index.ts
 /packages/server-core/src/index.ts
 /packages/gallery/www
+packages/social
```
Comment (HexaField) [inline .gitignore]: please remove this, seems to be a rebase error

File: .gitignore [hunk 2]
```diff
@@ -158,6 +158,7 @@ certs/
 /packages/projects/index.ts
 /packages/server-core/src/index.ts
 /packages/gallery/www
+packages/social/www
```
Comment (HexaField) [inline .gitignore:161]: please remove this, we no longer have the gallery and social packages

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/client-core/src/admin/common/hooks/Instance.hooks.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { useEffect } from 'react'
+
+export const useFetchAdminInstance = (user, adminInstanceState, InstanceService) => {
+  useEffect(() => {
+    if (user?.id.value != null && adminInstanceState.updateNeeded.value === true) {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/Instance.hooks.ts:5]: `user?.id.value && adminInstanceState.updateNeeded.value` would be same as above

File: packages/client-core/src/admin/common/hooks/Location.hooks.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react'
+
+export const useFetchLocation = (user, adminLocationState, adminScopeReadErrMsg, search, LocationService) => {
+  useEffect(() => {
+    if (user?.id?.value !== null && adminLocationState.updateNeeded.value && !adminScopeReadErrMsg?.value) {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/Location.hooks.ts:5]: `user?.id.value`

File: packages/client-core/src/admin/common/hooks/Location.hooks.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    }
+    // if (search) {
+    LocationService.fetchAdminLocations('increment', search)
+    // }
+  }, [search, user?.id?.value, adminLocationState.updateNeeded.value])
+}
+
+export const useFetchAdminScenes = (user: any, SceneService: any) => {
+  useEffect(() => {
+    if (user?.id.value != null) {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/Location.hooks.ts:16]: `user?.id.value`

File: packages/client-core/src/admin/common/hooks/Location.hooks.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      // && adminSceneState.scenes.updateNeeded.value === true) {
+      SceneService.fetchAdminScenes('all')
+    }
+  }, [user?.id?.value])
+}
+
+export const useFetchLocationTypes = (user, adminLocationState, LocationService) => {
+  useEffect(() => {
+    if (user?.id.value != null && adminLocationState.updateNeeded.value === true) {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/Location.hooks.ts:25]: `user?.id.value && adminLocationState.updateNeeded.value`

File: packages/client-core/src/admin/common/hooks/User.hooks.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { useEffect } from 'react'
+
+export const useFetchUsersAsAdmin = (user, adminUserState, UserService) => {
+  useEffect(() => {
+    if (user?.id.value != null && adminUserState.updateNeeded.value === true) {
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/hooks/User.hooks.ts:5]: user?.id.value && adminUserState.updateNeeded.value

File: packages/client-core/src/admin/services/InstanceService.ts [hunk 8]
```diff
@@ -52,22 +52,24 @@ export const useInstanceState = () => useState(state) as any as typeof state
...
-  fetchAdminInstances: async (incDec?: 'increment' | 'decrement') => {
+  fetchAdminInstances: async (incDec?: 'increment' | 'decrement', search?: string = '') => {
     const dispatch = useDispatch()
     {
       const skip = accessInstanceState().skip.value
       const limit = accessInstanceState().limit.value
       const user = accessAuthState().user
       try {
         if (user.userRole.value === 'admin') {
+          console.log(search)
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/services/InstanceService.ts:63]: Please remove this console log

File: packages/client-core/src/admin/services/LocationService.ts [hunk 9]
```diff
@@ -124,6 +116,30 @@ export const LocationService = {
...
+              name: 1
+            },
+            $skip: accessLocationState().skip.value,
+            $limit: accessLocationState().limit.value,
+            adminnedLocations: true
+          }
+        })
+        dispatch(LocationAction.locationsRetrieved(result))
+
+        console.log(result)
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/services/LocationService.ts:137]: please remove this console log

File: packages/client-core/src/admin/common/ConfirmModel.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  return (
+    <Dialog
+      open={popConfirmOpen}
+      onClose={handleCloseModel}
+      aria-labelledby="alert-dialog-title"
+      aria-describedby="alert-dialog-description"
+      classes={{ paper: classes.paperDialog }}
+    >
+      <DialogTitle id="alert-dialog-title">
+        Do you want to delete {label} <b>{name}</b>
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/common/ConfirmModel.tsx:28]: Please add a `?` at the end of the title.

Comment (kimenyikevin) [reply packages/client-core/src/admin/common/ConfirmModel.tsx:28]: Do you mean like thins ``name?`` ?

I am not able to get reason, I should do that

Comment (zulqarnainhanif) [reply packages/client-core/src/admin/common/ConfirmModel.tsx:28]: since it's a question there should be a question mark at the end of the title

**Review by zulqarnainhanif — APPROVED** [review]
The changes look fine! Have reviewed the code and QA'ed the feature as well.

**Comment by HexaField** [comment]
Thank you!

**Pull request merged**